### ISSUE TITLE
On some flights I get a constant msg “Connected to Stream Deck Plugin”

### ISSUE THREAD

**Issue by Virtual-Chris** [nguyenquyhy/Flight-Tracker-StreamDeck#56]
At some point mid flight, usually while adjusting altitude up or down the message pops up saying... and it won’t go away. It is annoying. What is causing this?
![17B3F893-95EF-4ECE-945A-EC9F11261544](https://user-images.githubusercontent.com/72712231/99360271-30c5a800-2865-11eb-90d4-26c42d48aa62.jpeg)

**Comment by craigusus** [comment]
Same here. Could there be an option to disable this? Or after X number of attempts to connect, the message disappears?
Even after closing the Stream Deck app down, the message still shows.

**Comment by nguyenquyhy** [comment]
I had implemented some throttling logic, but I haven't released that yet because I want to see if some other fixes would prevent this issues. Looks like it doesn't.

I'll release the throttling logic in the next update. Basically it will prevent the plugin to try to reconnect too frequently and allow SimConnect to recover.

**Comment by dflachbart** [comment]
Same here, running into the same issue all the time, most noticeably when you press and hold e.g. buttons like altitude to increase in 1000 ft steps instead of 100 ft etc. Also, even if I close Stream Deck and wait for the reconnect to 'catch up', the message stays on the screen for several minutes.

If you guys are able to fix this that would be awesome!

**Comment by iFred320** [comment]
Annoying. Please remove this.

**Comment by linmar22** [comment]
Duplicate closed, so responding here...

Is displaying the tooltip done by simconnect or the plugin? Because one of the solutions would just be to check when the last "Connected to Stream Deck plugin" tooltip was displayed and suppress any new ones until the timer expires. That way they can still do the reconnection dance in the background without bothering the user too much. Something like once every 5s would make it a lot less "in your face". 
Realize its a hack, but hey...

IMO End of the day - this would make more sense to if it was somehow displayed on the StreamDeck instead of MSFS, since MSFS works fine, but the StreamDeck can't communicate with it its disconnected.

**Comment by nguyenquyhy** [comment]
It already does something similar. Currently the reconnect is throttled at 2s, I might try to increase it to 5s or some exponential backoff.

I wouldn't say MSFS works fine in this case, unless you consider SimConnect not a part of MSFS. 
As I explained in the closed issue, the Stream Deck plugin always connects successfully to SimConnect (otherwise you won't even see the Connected message because that message itself is sent via SimConnect). However, SimConnect is experiencing some issue internally and stop reacting shortly after, and the only solution I can find is to force a disconnect (or user can also restart Stream Deck software), which is not a problem of the plug in either.

In other word, the problem is not that the plugin cannot connect to SimConnect, but SimConnect itself stops responding even though the connection is still there.

I don't prefer hiding the Connected message either. That doesn't help resolve this issue, and with that message at least everyone knows there is something wrong going on and restart Stream Deck software, which usually gives SimConnect enough time to self-recover.

**Comment by schuey97** [comment]
Having been using the plugin while flying the Baron G58 and the majority of the time have had no real issues with this message, only sometimes it will come up constantly for a few times and then go away.
However I have tried doing some flights in the Cessna Citation Longitude and the spamming of this messages happens a lot. Especially when power up the plane and when I came into land basically all of my buttons became in-active due to the "Connected to Stream Deck Plugin".....made for an interesting landing.
I have attached the logs from the Stream Deck from that last flight in case it helps.
[flightstreamdeck20210125.log](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/5866118/flightstreamdeck20210125.log)

Other than that, I'm really having fun with the plugin and its helping a lot with the flying. Keep up the great work.
